components/common/pdf-section.vue

```diff
@@ -0,0 +1,137 @@
+        title   : {type: String, required:true},
+        fileName: {type: String, required:true},
+    });
+
+    const emit = defineEmits(['onPdfDocument', 'onAfterPdf', 'onBeforeGetContent',
+                                'onBeforePdf']);
+
+    const {t}             = useI18n();
+    const { $recaptcha }  = useNuxtApp();
+
+    const isPdfing      = ref(false);
+    const userPdfHtml   = ref(null);
+
+    const downloadFileName = computed(()=>{
+        const fileName = props.fileName || props.title?.trim()?.replace(/[^a-z0-9]/gi, '-');
+
+        if(fileName.endsWith('.pdf'))
+            return fileName
+
+        return `${fileName}.pdf`;
+    });
+
+    const baseUrl = computed(()=>{
+
+        const domain = window.location.hostname.replace(/[^\.]+\./, '');
+        
+        if(domain=='localhost')
+            return 'https://ort.cbddev.xyz';
+
+        return window.location.origin;
+    })
+    async function onPdfDocument(){
+        emit('onPdfDocument');
+        
+        isPdfing.value = true;
+        await sleep(200);
+        userPdfHtml.value = document.querySelector(props.element).innerHTML;
+        await sleep(200);
+
+        const html = pdfThis('.cbd-user-pdf', {
+            debug:false,
+            pdfContainer:true,
+            importCSS:true,
+            importStyle : true,
+            // pageTitle : $('title').text(),
+            // loadCSS : '/app/css/pdf-friendly.css',
+        });	
+        // console.log(html);
+        try{
+            const captchaToken = await $recaptcha.getRecaptchaToken();
+            const res = await useAPIFetch('http://localhost:8000/api/v2017/generate-pdf', 
+                { 
+                    method:'POST', 
+                    body : { html }, 
+                    params : {
+                        'attachment-name' : downloadFileName.value,
+                        baseurl:baseUrl
+                    },
+                    responseType: "blob", 
+                    headers:{
+                        'x-captcha-v2-badge-token' : captchaToken
+                    }
+                })
+
+                // Create a Blob from the response data
+                const buffer = await res.arrayBuffer();
+                const pdfBlob = new Blob([buffer], { type: "application/pdf" });
+
+                // Create a temporary URL for the Blob
+                const url = window.URL.createObjectURL(pdfBlob);
+
+                // Create a temporary <a> element to trigger the download
+                const tempLink = document.createElement("a");
+                tempLink.href = url;
+                tempLink.setAttribute("download", downloadFileName.value);
+
+                // Append the <a> element to the body and click it to trigger the download
+                document.body.appendChild(tempLink);
+                tempLink.click();
+
+                // Clean up the temporary elements and URL
+                document.body.removeChild(tempLink);
```

Comment:
No need to add and remove. At least it works for me
https://github.com/scbd/kronos-web/blob/d9c09de2618a1dec747d8b17e12d1bbb8dba5138/filters/data-converter.js#L45 (Mine missing revokeObjectURL) !
```suggestion
                tempLink.click();

                // Clean up the temporary elements and URL
```

Reply:
done